Tests/SwiftLintFrameworkTests/ConfigurationTests+MultipleConfigs.swift

```diff
@@ -411,12 +411,159 @@ extension ConfigurationTests {
+        _ parentConfiguration: Configuration?,
+        configurations: [Configuration],
+        ruleType: any Rule.Type
+    ) {
+        for configuration in configurations {
+            if case .default(let disabledRules, let optInRules) = configuration.rulesMode {
+                let mergedConfiguration = parentConfiguration?.merged(withChild: configuration) ?? configuration
+                let isEnabled = mergedConfiguration.contains(rule: ruleType)
+                let issue = validateConfiguredRuleIsEnabled(
+                    parentConfiguration: parentConfiguration,
+                    disabledRules: disabledRules,
+                    optInRules: optInRules,
+                    ruleType: ruleType
+                )
+                XCTAssertEqual(isEnabled, issue == nil)
+                if let issue {
+                    let ruleIdentifier = ruleType.identifier
+                    if disabledRules.isEmpty, optInRules.isEmpty {
+                        if parentConfiguration == nil ||
+                            parentConfiguration == Configuration.emptyDefaultConfiguration() {
+                            XCTAssertEqual(issue, Issue.ruleNotEnabledInOptInRules(ruleID: ruleIdentifier))
+                            continue
+                        }
+
+                        if parentConfiguration == Configuration.emptyOnlyConfiguration() {
+                            if ruleType is any OptInRule.Type {
+                                XCTAssertEqual(issue, Issue.ruleNotEnabledInOptInRules(ruleID: ruleIdentifier))
+                            } else {
+                                XCTAssertEqual(issue, Issue.ruleNotEnabledInParentOnlyRules(ruleID: ruleIdentifier))
+                            }
+                            continue
+                        }
+                        if parentConfiguration == Configuration.optInDisabledDefaultConfiguration(ruleIdentifier) ||
+                            parentConfiguration == Configuration.disabledDefaultConfiguration(ruleIdentifier) {
+                            XCTAssertEqual(issue, Issue.ruleDisabledInParentConfiguration(ruleID: ruleIdentifier))
+                            continue
+                        }
+                    }
+                    XCTAssertEqual(issue, Issue.ruleDisabledInDisabledRules(ruleID: ruleIdentifier))
+                }
+            }
+        }
+    }
+
+    func testOnlyConfigurationDisabledRulesWarnings() {
+        let optInRuleType = ImplicitReturnRule.self
+        XCTAssertTrue((optInRuleType as Any) is any OptInRule.Type)
+        testOnlyConfigurationDisabledRulesWarnings(ruleType: optInRuleType)
+
+        let defaultRuleType = BlockBasedKVORule.self
+        XCTAssertFalse((defaultRuleType as Any) is any OptInRule.Type)
+        testOnlyConfigurationDisabledRulesWarnings(ruleType: defaultRuleType)
+    }
+
+    private func testOnlyConfigurationDisabledRulesWarnings(ruleType: any Rule.Type) {
```

Comment:
I suggest to inline this method and unroll all the test cases. The lists and loops make these rather simple cases harder to follow.

Reply:
yep, much shorter now